api/envoy/extensions/filters/http/oauth2/v3/oauth.proto

```diff
@@ -62,8 +62,17 @@ message OAuth2Credentials {
 
 // OAuth config
 //
-// [#next-free-field: 11]
+// [#next-free-field: 12]
 message OAuth2Config {
+  enum AuthType {
+    // The client_id and client_secret will be sent using HTTP Basic authentication scheme.
+    BASIC_AUTH = 0;
```

Comment:
following up on what @htuch said, we probably want the default value to be a no-op (e.g.: `NO_AUTH` or such).

Reply:
@rgs1 @htuch, I have made the existing behaviour as the default auth method (`URL_ENCODED_BODY`).